src/components/manage-registry/manage-agents/ManageAgents.tsx

```diff
@@ -118,9 +179,9 @@ export function ManageAgents({
         <Flex direction="row" gap="2">
           <Input
             width="70%"
-            value={newUserAgentAddress}
-            placeholder="User agent address"
-            onChange={(e) => setNewUserAgentAddress(e.target.value)}
+            size="md"
+            ref={inputRef}
```

Comment:
Why switch from controlled input and storing value in the state to using ref and reading directly?